### PR TITLE
Fix: Display only data from intersection of DB and ElasticSearch

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

- {List all applicable Jira tickets}

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/src/repositories/submission-repository.ts [hunk 1]
```diff
@@ -376,9 +380,37 @@ export class SubmissionRepository extends BaseRepository {
...
+    const sqlStatement = SQL`
+      SELECT
+        uuid
+      FROM
+        submission
+      WHERE
+        record_end_date IS NULL;
+    `;
+
+    const response = await this.connection.sql<{ uuid: string }>(sqlStatement);
```
Comment (NickPhura) [inline api/src/repositories/submission-repository.ts]: You can also use your IDatasetId interface here: `.sql<IDatasetId>`

File: api/src/services/submission-service.ts [hunk 2]
```diff
@@ -162,10 +163,20 @@ export class SubmissionService extends DBService {
...
     return this.submissionRepository.getSubmissionRecordJSONByDatasetId(datasetId);
   }
 
+  /**
+   *
+   *
+   * @return {*}  {Promise<string>}
+   * @memberof SubmissionService
+   */
+  async getDatasetIdsFromDB(): Promise<IDatasetId[]> {
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: If we are no longer using this, lets get rid of it (and whatever other related code)

File: api/src/repositories/submission-repository.ts [hunk 3]
```diff
@@ -48,6 +48,10 @@ export interface ISubmissionModel {
 export interface ISubmissionModelWithStatus extends ISubmissionModel {
   submission_status: string;
 }
+
+export interface IDatasetId {
```
Comment (curtisupshall) [inline api/src/repositories/submission-repository.ts]: Should we consider a different name for this interface?

**Review by curtisupshall — DISMISSED** [review]
Looks good to me. My single comment is a nit-pick

File: api/src/repositories/submission-repository.ts (hunk 3 above)
Comment (curtisupshall) [reply api/src/repositories/submission-repository.ts]: Namely I wouldn't expect an object of type "dataset ID" to have a property "uuid" which actually contains said ID.

Comment (NickPhura) [reply api/src/repositories/submission-repository.ts]: Good point, though this is going to be removed, so easy fix haha

edit: removed in recent commit

**Review by NickPhura — APPROVED** [review]

**Pull request merged**